cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/CreateSchedulesStage.java

```diff
@@ -103,5 +113,19 @@ public void process(ApplicationWithPrograms input) throws Exception {
 
   private void createSchedule(ProgramId programId, ScheduleSpecification scheduleSpec) throws SchedulerException {
     scheduler.schedule(programId, scheduleSpec.getProgram().getProgramType(), scheduleSpec.getSchedule());
+    addProgramSchedule(programId, scheduleSpec);
+  }
+
+  private void addProgramSchedule(ProgramId programId, ScheduleSpecification scheduleSpec) {
+    Schedule schedule = scheduleSpec.getSchedule();
+    if (schedule instanceof TimeSchedule) {
+      ProgramSchedule programSchedule =
+        Schedulers.getProgramSchedule((TimeSchedule) schedule, programId);
+      try {
+        programScheduler.addSchedule(programSchedule);
+      } catch (AlreadyExistsException e) {
+        LOG.warn("ProgramSchedule {} already exists for program {}", schedule.getName(), programId);
```

Comment:
It seems that this will now always happen when an app is updated. If its schedules don't change, are we gonna try to create the program schedule and then log this warning? That is misleading because there is really nothing wrong. 

It may be an error if the schedule exists for a different program. Not sure. But then it is probably an error, not a warning.